native/core/src/execution/operators/scan.rs

```diff
@@ -200,6 +217,21 @@ impl ScanExec {
 
         let mut env = JVMClasses::get_env()?;
 
+        let mut timer = jvm_fetch_time.timer();
+
+        let num_rows: i32 = unsafe {
+            jni_call!(&mut env,
+        comet_batch_iterator(iter).has_next() -> i32)?
+        };
```

Comment:
Ideally, I would have had `has_next` return a `bool` rather than `i32` but I could not figure out how to make that happen with our jni macros. I am open to suggestions on how to improve this.

Reply:
I don't see a `has_next` in CometBatchIterator.java. Should this be `comet_batch_iterator(iter).next()` ? 
And if you want to change the return value you can make the change in the Java class and also in the macro invocation.